kedro-datasets/kedro_datasets/ibis/table_dataset.py

```diff
@@ -0,0 +1,124 @@
+"""Provide data loading and saving functionality for Ibis's backends."""
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import TYPE_CHECKING, Any, ClassVar
+
+import ibis.expr.types as ir
+from kedro.io import AbstractDataset, DatasetError
+
+if TYPE_CHECKING:
+    from ibis import BaseBackend
+
+
+class TableDataset(AbstractDataset[ir.Table, ir.Table]):
+    """``TableDataset`` loads/saves data from/to Ibis table expressions.
+
+    Example:
+
+    .. code-block:: pycon
+
+        >>> from kedro_datasets.ibis import TableDataset
```

Comment:
Would it be possible to add an local example here with some dummy dataframe / duckdb? I know there is a blog post, but just looking at the docs/docstring I think it's not that easy to get how an user should use this.

Reply:
Ideally we'd have both a `yaml` and `python` example.

Reply:
Just fyi, when connecting to ibis there are two major "paradigms", either a filebased connection or a db connection. From my limited knowledge I think the configuration arguments are quite different.

For reference, I am currently connecting to mssql dbs like this:

```python
conn_table = TableDataset(
    connection={
        "host": "xxx.sql.azuresynapse.net",
        "database": "xxx",
        "query": {"driver": "ODBC Driver 17 for SQL Server"},
        "backend": "mssql",
    },
    table_name="xxx",
    load_args={"schema": "xxx"},
)
```

Reply:
the documentation here should probably clarify what the different arguments map to in the ibis connection object. 

in the case of a `filepath` being provided, it will map to the backend method `read_{file_format}`, e.g. https://ibis-project.org/backends/duckdb#ibis.backends.duckdb.Backend.read_parquet

otherwise it will get the table from `ibis.{backend}.Backend.table` where the Backend object is obtained through `ibis.{backend}.connect` e.g. https://ibis-project.org/backends/mssql

Reply:
Sorry, I forgot to add an example, it seems. 😅